### DISCUSSION TITLE
CSS Modules - tailwindcss-animate

### DISCUSSION THREAD

**Discussion by mverissimo** [tailwindlabs/tailwindcss#11164] [Help] [answered]
Hi, I'm trying to use this [library](https://github.com/jamiebuilds/tailwindcss-animate) with CSS modules but the keyframes are not injected into the head styles.

I can't find any problem with the plugin.

[reproduction](https://codesandbox.io/p/sandbox/css-modules-vczjci?file=%2Fpackage.json&selection=%5B%7B%22endColumn%22%3A24%2C%22endLineNumber%22%3A20%2C%22startColumn%22%3A24%2C%22startLineNumber%22%3A20%7D%5D)

**Comment by wongjn** [comment, marked as answer]
It seems like the `.animation-in` and `.animation-out` rules defined by that library are added using `addUtilities()` which means the `@keyframes` are not associated with them. It seems like Tailwind needs to know whether the an `@keyframes` definition is used by an `animation-*` class:

https://github.com/tailwindlabs/tailwindcss/blob/4893cad29c581d5e557f75ebad999d8f343d61b4/src/corePlugins.js#L915-L945

Here's a modified version of the library where the animation in the CSS module works: https://stackblitz.com/edit/vitejs-vite-zdvzjy?file=animate.js

Consider raising an issue with the library.

**Reply by mverissimo** [reply]
You're amazing, thanks!

Has open [issue](https://github.com/jamiebuilds/tailwindcss-animate/issues/25), consider opening a pull request, what do you think?
